controllers/scopeinstance_controller.go

```diff
@@ -81,40 +171,131 @@ func (r *ScopeInstanceReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 		for _, namespace := range in.Spec.Namespaces {
 			for _, cr := range st.Spec.ClusterRoles {
 				objects = append(objects, &rbacv1.RoleBinding{
 					ObjectMeta: metav1.ObjectMeta{
 						GenerateName: cr.GenerateName + "-",
 						Namespace:    namespace,
+						Labels: map[string]string{
+							"scopeInstance": "operators.io.operator-framework-v1",
+						},
+						// OwnerReferences: []metav1.OwnerReference{{
+						// 	APIVersion: in.APIVersion,
+						// 	Kind:       in.Kind,
+						// 	Name:       in.GetObjectMeta().GetName(),
+						// 	UID:        "d9607e19-f88f-11e6-a518-42010a800195",
+						// }},
 					},
 					Subjects: cr.Subjects,
 					RoleRef: rbacv1.RoleRef{
 						Kind: "ClusterRole",
 						Name: cr.GenerateName,
 					},
 				})
+
 			}
 		}
-
 	}
 
+	return objects, nil
+}
+
+func (r *ScopeInstanceReconciler) createRBAC(ctx context.Context, objects []client.Object) error {
+	// Create Objects
 	for _, object := range objects {
-		log.Log.Info("Creating object", "object", object)
+		r.logger.Debug("Creating object", "object", object)
 		if err := r.Client.Create(ctx, object); err != nil {
+			if k8sapierrors.IsAlreadyExists(err) {
+				r.logger.Debug("RBAC is already Present", "object", object.GetName())
+				continue
+			}
+
 			log.Log.Error(err, "Error creating object", "object", object)
+			return err
 			/// TODO: try to create all the (cluster)roleBindings before returning an aggregate error.
 		}
 	}
+	return nil
+}
+
+func (r *ScopeInstanceReconciler) getRBACForDeletion(clusterRoleBindingMap map[string]rbacv1.ClusterRoleBinding, roleBindingMap map[string]rbacv1.RoleBinding) ([]client.Object, error) {
+
+	deleteObjects := []client.Object{}
+	// below is for cleanup, check if Map(Key, value) of Role Binding is empty or not if not then those
+	// are remaining ones which we have to delete
+	for key, element := range roleBindingMap {
+		split := strings.Split(key, "-")
+
+		deleteObjects = append(deleteObjects, &rbacv1.RoleBinding{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      element.GetName(),
+				Namespace: split[0],
+			},
+		})
+	}
 
-	log.Log.Info("No ScopeInstance error")
+	// below is for cleanup, check if Map(Key, value) of Cluster Role Binding is empty or not if not then those
+	// are remaining ones which we have to delete
+	for _, element := range clusterRoleBindingMap {
+		deleteObjects = append(deleteObjects, &rbacv1.ClusterRoleBinding{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: element.GetName(),
+			},
+		})
+	}
 
-	return ctrl.Result{}, nil
+	return deleteObjects, nil
+}
+
+func (r *ScopeInstanceReconciler) deleteRBAC(ctx context.Context, deleteObjects []client.Object) error {
+	// Create Objects
+	// Delete Unnecessary Objects
+	for _, object := range deleteObjects {
+		r.logger.Debug("Deleting object", "object", object)
+		if err := r.Client.Delete(ctx, object); err != nil {
+			log.Log.Error(err, "Error deleting object", "object", object)
```

Comment:
No need to log errors if they're going to be surfaced anyways.

Reply:
removed log line.